doc/extdev/nodes.rst

```diff
@@ -48,6 +48,10 @@ argument `_sig_element=True` of :py:class:`desc_sig_element`, e.g.:
 
       class desc_custom_sig_node(desc_sig_element, _sig_element=True): ...
 
+This automatically adds the custom node class to :py:data:`SIG_ELEMENTS`. For
+backwards compatibility, it is still possible to add the nodes directly using
+``SIG_ELEMENTS.add(desc_custom_sig_node)``.
```

Comment:
This is still saying *what* rather than *why*. Why should an extension developer care that a new node is in SIG_ELEMENTS?

Reply:
By design, if the translator class is orthogonal to `SphinxTranslator` (i.e., it directly inherits `docutils.nodes.NodeVisitor`), the translator should support all `desc_sig_*` like nodes or a generic `visit_desc_sig_element`. So, if an extension developer actually uses a custom translator and implement custom `desc_sig_*` visitor methods, they should be "notified" that a new node exists; if not, the nodes are automatically transformed to inline nodes without the developer being aware of. 

Actually, should I rather remove the paragraph on `SIG_ELEMENTS` and only expose what the current `desc_sig_*` nodes are present? and put the thing about `SIG_ELEMENTS` directly at the level of `sphinx.addnodes` instead? I think it's more likely to have a custom translator rather than having a custom `desc_sig_*` node. If *we* add new nodes, we would update `extdev/nodes.rst`, but if an extension developer wants to add new `desc_sig_*` nodes, they should know that they need to add them to `SIG_ELEMENTS` as well, so having the documentation at the level of `desc_sig_*` and `SIG_ELEMENTS` may be more relevant.